main.c

```diff
@@ -245,9 +247,41 @@ int NMEA_message_handler(int sock)
 * @date 17.04.2014 born
 *
 */ 
-void pressure_measurement_handler(void)
+void pressure_measurement_handler(int sock)
 {
 	static int meas_counter = 1;
+	static int temp_counter = 0;
+	char s[70];
+	int result;
+	int sock_err = 0;
+	
+	if (temp_sensor.active) {
+	   if (temp_counter==0) {
+	      OWReset(&temp_sensor);// Reset
+	      OWWriteByte(&temp_sensor,0xCC); // Skip ROM
+	      OWWriteByte(&temp_sensor,0x44); // Conversion
+	      temp_counter=1;
+	   } else 
+	   if (temp_counter == temp_sensor.rollover) {
+	     OWReset(&temp_sensor);  // Reset 
+	     OWWriteByte(&temp_sensor,0xCC); // Skip ROM
+	     OWWriteByte(&temp_sensor,0xBE); //Read Scratchpad
+	     OWReadTemperature(&temp_sensor); // Convert output to temperature
+	     temp_counter=0;
+	     if ((temp_sensor.valid==1) && (config.output_POV_T == 1 )) {
+	       // Compose POV slow NMEA sentences
+	       result = Compose_Temperature_POV(&s[0], temp_sensor.temperature);
```

Comment:
I looked over the ds2482 branch and while there are things I like about it more than my version (its definitely a little cleaner), there are also things I don't like about it -- in fact, based on my understanding of how the DS18B20 works, I'm surprised it works at all.

First, there is limited if any error checking.  My version could be improved, and I'd like to do that, but the ds2482 branch is worse.

Second, from what I can tell, it executes the pressure_measurement_handler function every 12.5ms.  On every 4th time through (50ms) it launches a temperature conversion.  It reads out the temperature measurement the next time through (12.5ms after the conversion request).  The fastest conversion time is 93.75ms, so it's always going to be initiating a new conversion before the previous conversion completes, and is always reading it out before valid data is available.

Mine on the other hand allows you to configure how much resolution you want (a feature the ds2482 branch seems to lack) and then waits the appropriate amount of time before reading out the measurement and launching a new conversion.  The reason I'm composing and writing out the NMEA sentence in temperature_measurement_handler is because I am composing the sentence AFTER valid data is available.  

Also, in my new version, I only do anything with the ds2482 if the config_POV_T switch is set.  I'm not sure why we would ever want to collect information from the sensors but not output it.